### PR TITLE
Completion improvements

### PR DESCRIPTION
**Pull request by jmbockhorst**
### Improved sorting of qualified imports and auto imports. Ref #287.
The problem is that if you have `Html` imported and you type `div`, there are 2 completions: `div (auto imports)` and `Html.div`. The auto imported will always be first because it matches the string better (in VSCode at least). To fix this, I check if the typed text matches the function name and add filter text to `Html.div` completion. So if we type `di`, `Html.div` gets a filter text of `div`, and thus will sort to the top. If we start to type `Ht`, the filter text will just be the label, `Html.div`. This is the main fix for #287, but there are still some cases I want to look at later.

### Added completions for module values or possible submodules
If a module is imported, it now should all possible values for that module. Ex: `Html.{-caret-}` will show `a, div, span, ...`. It also shows possible submodules (not imported). In the case: `Attributes` or `Styled.Attributes` could also show up as completions. 
 
### Fixed record completion interfering with module completions. Ref #288.   
Reverts #289 with a proper fix. Instead or checking for all `field_access_segment` descendents, it now specifically looks at the target word. 

I also refactored `CompletionProvider` to use object parameters instead of function parameters. It made it more clear and easier to pass in optional parameters (like `filterText`).

And of course, added a lot of tests to cover these cases.

### CODE REVIEW

**Comment by jmbockhorst** [comment]
Also pushed a fix as a follow up to 71cad988fe31243606d8dbe4231023727f1836f6 where `element.replace(".", "/")` didn't replace all the dots. This caused paths like `Module/Submodule.AnotherSub`

**Comment by Razzeee** [comment]
Do you know why `replace` isn't enough? I'm not sure I understand what's going on.

**Comment by jmbockhorst** [comment]
Because it only replaces the first match. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace

**Comment by Razzeee** [comment]
Oh crap thank you. Not sure who designed that api, but I would have never expected that.

File: src/providers/completionProvider.ts [hunk 1]
```diff
@@ -219,11 +288,17 @@ export class CompletionProvider {
         ...this.findDefinitionsForScope(nodeAtPosition, tree, replaceRange),
       );
 
+      const currentTarget = targetLine.substring(
```
Comment (Razzeee) [inline src/providers/completionProvider.ts:291]: Isn't this identical to `targetWord`?

Comment (jmbockhorst) [reply src/providers/completionProvider.ts:291]: Yeah I think so. I added this first then targetWord later, so should be good to get rid of this one.

**Pull request merged**